src/Microsoft.Azure.SignalRService/ServiceManager.cs

```diff
@@ -3,16 +3,53 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Security.Claims;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.Azure.SignalRService
 {
     public class ServiceManager : IServiceManager
     {
+
+        internal ServiceManagerContext ServiceManagerContext { get; }
+
+        internal ServiceManager(ServiceManagerContext serviceManagerContext)
+        {
+            ServiceManagerContext = serviceManagerContext;
+        }
+
         public Task<IServiceHubContext> CreateHubContextAsync(string hubName)
         {
-            throw new NotImplementedException();
+            var hubInfo = new HubInfo(hubName);
+
+            var serviceCollection = new ServiceCollection();
+            serviceCollection.AddSignalRCore();
+
+            // remove default hub lifetime manager
+            var serviceDescriptor = serviceCollection.FirstOrDefault(descriptor => descriptor.ServiceType == typeof(HubLifetimeManager<>));
+            serviceCollection.Remove(serviceDescriptor);
+
+            switch (ServiceManagerContext.ServiceManagerOptions.ServiceTransportType)
+            {
+                case ServiceTransportType.Persistent:
+                    {
+                        var restHubLifetimeManager = new RestHubLifetimeManager(ServiceManagerContext, hubInfo);
+                        serviceCollection.AddSingleton(typeof(HubLifetimeManager<Hub>), restHubLifetimeManager);
```

Comment:
What will happen if you call CreateHubContextAsync multiple times?

Reply:
They will always send to the same hub for different ServiceHubContext. Thus, service provider should created for each ServiceHubContext.

Reply:
ServiceHubContexts are independent to each other since service collections are independent to each other. When you call CreateHubContextAsync, a new service collection is created.